### PR TITLE
[websocketpp] Update to 0.8.2

### PR DESCRIPTION
**Pull request by c72578**
The vcpkg port of websocketpp has been pointing to the upstream commit
zaphoyd/websocketpp@1c699ce46843a787482a703fdeff9271fbb7bb5d of a PR, in order to fix build issues of websocketpp 0.8.1 with
boost 1.70 (#6141). This fix has been merged upstream in the meantime
and is part of the release 0.8.2 now.

- Update to WebSocket++/0.8.2 release
- Remove deprecated `include(vcpkg_common_functions)`
- Use `${PORT}` instead of `websocketpp`

**Describe the pull request**

- What does your PR fix? Fixes issue #
Use new upstream release 0.8.2 of websocketpp
Changelog: https://github.com/zaphoyd/websocketpp/blob/master/changelog.md
Info from testing the updated package: All installed files of the vcpkg package of websocketpp stay identical, except version.hpp (0.8.1->0.8.2)
- Which triplets are supported/not supported? Have you updated the CI baseline?
All triplets are supported.
CI baseline: OK, no update required
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/websocketpp/portfile.cmake [hunk 1]
```diff
@@ -1,25 +1,24 @@
...
+    SHA512 f185a66e5a7c783254352a6ef87e2e559f681032b7368765d08393ed12bcae76825abed7dcaea73de09df644320409dad46279701f5f469520542a2c9b6a6163
     HEAD_REF master
 )
 
-file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/share/websocketpp)
+file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/share/${PORT})
 
 # Copy the header files
-file(COPY "${SOURCE_PATH}/websocketpp" DESTINATION "${CURRENT_PACKAGES_DIR}/include" FILES_MATCHING PATTERN "*.hpp")
+file(COPY "${SOURCE_PATH}/${PORT}" DESTINATION "${CURRENT_PACKAGES_DIR}/include" FILES_MATCHING PATTERN "*.hpp")
```
Comment (LilyWangL) [inline ports/websocketpp/portfile.cmake:14]: Thanks for your PR. Could you please revert this change? The directory in ${SOURCE_PATH} should be keep original name.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/websocketpp/portfile.cmake (hunk 1 above)
Comment (c72578) [reply ports/websocketpp/portfile.cmake:14]: @LilyWangL Thanks for your review. Could you please give a short explanation, why `${PORT}` should not be used in this particular case?

Comment (LilyWangL) [reply ports/websocketpp/portfile.cmake:14]: The `${SOURCE_PATH}` and its subdirectory are extracted from the source code to vcpkg. They are not created by VCPKG. So I think they should keep original name.

Comment (c72578) [reply ports/websocketpp/portfile.cmake]: OK, thanks for the feedback. The commit has been amended according to your suggestion.

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR!
